ReSwift/CoreTypes/Store.swift

```diff
@@ -28,7 +28,7 @@ open class Store<State: StateType>: StoreType {
             subscriptions.forEach {
                 // if a selector is available, subselect the relevant state
                 // otherwise pass the entire state to the subscriber
-                $0.subscriber?._newState(state: $0.selector?(state) ?? state)
+                $0.subscriber?._newState(state: $0.selector?(state) ?? state!)
```

Comment:
@tonyarnold actually it seems my change here is wrong. Your comment prompted me to look at little farther into the behaviour of the app and it appears it should be doing `?? state as Any`

I also think theres a larger issue of https://github.com/ReSwift/ReSwift/pull/177 changing reducer to allow returning nil, but that not getting propagated to subscribers.

I'll update my PR